src/olympia/activity/views.py

```diff
@@ -50,17 +53,23 @@ def check_object_permissions(self, request, obj):
 
     def get_serializer_context(self):
         ctx = super(VersionReviewNotesViewSet, self).get_serializer_context()
-        ctx['to_highlight'] = self.pending_queryset(
-            amo.LOG.DEVELOPER_REPLY_VERSION)
+        ctx['to_highlight'] = filter_queryset_to_pending_replies(
+            self.get_queryset())
         return ctx
 
-    def pending_queryset(self, log_type):
-        version_qs = self.get_queryset()
-        latest_reply = version_qs.filter(action=log_type.id).first()
-        if not latest_reply:
-            return version_qs
-        return version_qs.filter(created__gt=latest_reply.created)
-
+    def create(self, request, *args, **kwargs):
+        version = self.get_version_object()
+        if not version == version.addon.latest_or_rejected_version:
+            raise ParseError(
+                'Only latest versions of addons can have notes added.')
+        activity_object = log_and_notify(
+            action_from_user(request.user, version), request.data['comments'],
+            request.user, version)
+        serializer = self.get_serializer(activity_object)
+        response = Response(serializer.data)
+        if response.status_code == 200:
+            response.status_code = 201
```

Comment:
This looks weird, what are you trying to do here? I think something like

```python
from rest_framework import status
serializer.is_valid(raise_exception=True)
return Response(serializer.data, status_code=status.HTTP_201_CREATED)
```
could work too? This will return a `400` response with validation errors right away and in case of success a `201 CREATED` response.